src/components/pages/profile/contribution/EditManualContribution.vue

```diff
@@ -0,0 +1,149 @@
+      :items="contributionTypes"
+      class="mb-3"
+    />
+
+    <AppInput
+      v-model="contribution.amount"
+      label="Amount"
+      type="number"
+      class="mb-3"
+    />
+
+    <AppRadioGroup
+      v-if="showChangePeriod"
+      v-model="contribution.period"
+      name="period"
+      label="Period"
+      :options="[
+        ['monthly', 'Monthly'],
+        ['yearly', 'Yearly'],
+      ]"
+      class="mb-4"
+    />
+
+    <AppSelect
+      v-model="contribution.source"
+      label="Source"
+      :items="sourceOptions"
+      class="mb-3"
+    />
+
+    <AppInput v-model="contribution.reference" label="Reference" class="mb-3" />
+
+    <AppButton
+      :disabled="validation.$invalid"
+      type="submit"
+      variant="link"
+      class="mt-6"
+      :loading="loading"
+    >
+      {{ t('contribution.updateContribution') }}
+    </AppButton>
+  </form>
+</template>
+<script lang="ts" setup>
+import { computed, reactive, ref, toRef, watch } from 'vue';
+import { useI18n } from 'vue-i18n';
+import useVuelidate from '@vuelidate/core';
+import AppButton from '../../../forms/AppButton.vue';
+import AppInput from '../../../forms/AppInput.vue';
+import AppSelect from '../../../forms/AppSelect.vue';
+import AppRadioGroup from '../../../forms/AppRadioGroup.vue';
+import {
+  ContributionPeriod,
+  ContributionType,
+  MembershipStatus,
+} from '@beabee/beabee-common';
+import {
+  fetchMember,
+  forceUpdateContribution,
+} from '../../../../utils/api/member';
+import AppHeading from '../../../AppHeading.vue';
+
+const validation = useVuelidate();
+
+const { t } = useI18n();
+
+const props = defineProps<{
+  id: string;
+}>();
+
+const contribution = reactive({
+  type: 'manual',
+  amount: 0,
+  period: ContributionPeriod.Monthly,
+  source: '',
+  reference: '',
+});
+
+const contributionTypes = [
+  {
+    id: 'None',
+    label: 'None',
+  },
+  {
+    id: 'Manual',
+    label: 'Manual',
+  },
+];
+
+const sourceOptions = [
+  {
+    id: 'none',
+    label: 'None',
+  },
+  {
+    id: 'cheque',
+    label: 'Cheque',
+  },
+];
+
+const loading = ref(false);
+
+const isActiveMember = computed(
+  () => contribution.membershipStatus === MembershipStatus.Active
+);
+const isManualActiveMember = computed(
+  () => isActiveMember.value && contribution.type === ContributionType.Manual
+);
+
+// Only non-active members and monthly manual contributors can change their period
+// as otherwise proration gets complicated
+const showChangePeriod = computed(
+  () =>
+    !isActiveMember.value ||
+    (isManualActiveMember.value &&
+      contribution.period !== ContributionPeriod.Annually)
+);
+
+watch(
+  toRef(props, 'id'),
+  async (id) => {
+    const member = await fetchMember(id, ['contribution']);
+    contribution.type = member.contribution.type;
+    contribution.amount = member.contribution.amount;
+    contribution.period = member.contribution.period;
+    contribution.source = member.contribution.paymentSource.source;
```

Comment:
`paymentSource` can be different things, so the compiler needs to be sure it's a `ManualPaymentSource` before giving you access to `source` and `reference`

The definition of the type is:
```ts
export type PaymentSource =
  | StripeCardPaymentSource
  | GoCardlessDirectDebitPaymentSource
  | StripeBACSPaymentSource
  | StripeSEPAPaymentSource
  | ManualPaymentSource;
```
`PaymentSource` is a [discriminated union](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions), which means it has a property which tells you which of the possible subtypes it actually is. 

In this case that property is called `method`, you can see this because all the types listed have a unique value for `method` (see [this bit in api.interface.ts](https://github.com/beabee-communityrm/beabee-frontend/blob/feat/contact-contrib/src/utils/api/api.interface.ts#L100-L139)). So to be sure that `paymentMethod` is a `ManualPaymentSource` you need to check if `method` is `null`:
```ts
const paymentSource = member.contribution.paymentSource;
if (paymentSource?.method === null) {
  contribution.source = paymentSource.source || '';
  contribution.reference = paymentSource.reference || '';
} else {
  // In practice this shouldn't be possible as you've checked in the type is Manual or None above
  contribution.source = '';
  contribution.reference = '';
}